pkg/pipeline/stream/processor.go

```diff
@@ -42,40 +55,59 @@ func (n *ProcessorNode) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-
 	defer cleanup()
+
+	n.logger.Debug(ctx).Msg("opening processor")
+	err = n.Processor.Open(ctx)
+	if err != nil {
+		n.logger.Err(ctx, err).Msg("failed opening processor")
+		return cerrors.Errorf("couldn't open processor: %w", err)
+	}
+
+	defer func() {
+		n.logger.Debug(ctx).Msg("tearing down processor")
+		tdErr := n.Processor.Teardown(ctx)
+		err = cerrors.LogOrReplace(err, tdErr, func() {
+			n.logger.Err(ctx, tdErr).Msg("could not tear down processor")
+		})
+	}()
+
 	for {
 		msg, err := trigger()
 		if err != nil || msg == nil {
 			return err
 		}
 
 		executeTime := time.Now()
-		rec, err := n.Processor.Process(msg.Ctx, msg.Record)
+		recsIn := []opencdc.Record{msg.Record.ToOpenCDC()}
+		recsOut := n.Processor.Process(msg.Ctx, recsIn)
 		n.ProcessorTimer.Update(time.Since(executeTime))
-		if err != nil {
-			// Check for Skipped records
-			switch err {
-			case processor.ErrSkipRecord:
-				// NB: Ack skipped messages since they've been correctly handled
-				err := msg.Ack()
-				if err != nil {
-					return cerrors.Errorf("failed to ack skipped message: %w", err)
-				}
-			default:
-				err = msg.Nack(err, n.ID())
-				if err != nil {
-					return cerrors.Errorf("error executing processor: %w", err)
-				}
-			}
-			// error was handled successfully, we recovered
-			continue
+
+		if len(recsIn) != len(recsOut) {
+			return msg.Nack(
+				cerrors.Errorf("processor was given %v record(s), but returned %v", len(recsIn), len(recsOut)),
+				n.ID(),
+			)
```

Comment:
If we have a DLQ configured, this will return `nil` and the node will stop without stopping the pipeline. We need to make sure we return an error so a stop of the pipeline is truggered. Something like this:
```suggestion
			err = cerrors.Errorf("processor was given %v record(s), but returned %v", len(recsIn), len(recsOut))
			if nackErr := msg.Nack(err, n.ID()); nackErr != nil {
				return nackErr
			}
			return err
```